### PR TITLE
ENG-5042 implement uninstall in mocked state

### PR DESCRIPTION
**Pull request by ichalagashvili**
Implements uninstall modal for different scenarios and cases (scenarios can be checked in the ticket)

### CODE REVIEW

File: src/ui/component-repository/components/item/install-controls/ComponentUninstallManagerModal.js [hunk 1]
```diff
@@ -1,32 +1,63 @@
...
 import ComponentUninstallStart from 'ui/component-repository/components/item/install-controls/ComponentUninstallStart';
+import { componentUninstallOngoingProgress, setInstallUninstallProgress, setSelectedECRComponent, uninstallECRComponent } from 'state/component-repository/components/actions';
 
 const ComponentUninstallManagerModal = () => {
   const componentUsageList = useSelector(getComponentUsageList);
-  const usageDataLoading = useSelector(getLoading)['component-repository/component-usage'];
   const selectedEcrComponent = useSelector(getECRComponentSelected);
+  // const componentUninstallStatus =
+  // useSelector(state => getECRComponentUninstallStatus(state,
+  // { component: selectedEcrComponent }));
```
Comment (DMinnai) [inline src/ui/component-repository/components/item/install-controls/ComponentUninstallManagerModal.js]: Can we deleted unused code?

File: src/ui/component-repository/components/item/install-controls/ComponentUninstallManagerModal.js [hunk 2]
```diff
@@ -1,32 +1,63 @@
...
+  const lastInstallApiResponse = useSelector(state =>
+    getECRComponentLastInstallApiResponsePayload(state, { component: selectedEcrComponent })) || {};
+  const { progress, status: componentUninstallStatus } = lastInstallApiResponse;
+  // const progress = useSelector(getInstallUninstallProgress);
```
Comment (DMinnai) [inline src/ui/component-repository/components/item/install-controls/ComponentUninstallManagerModal.js]: same here

File: src/ui/component-repository/components/item/install-controls/ComponentUninstallProcessState.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+          }}
+          />
+        </div>
+        <div className="ComponentUninstallProcessState__in-progress-body-progress-footer-wrapper">
+          <span className="ComponentUninstallProcessState__in-progress-body-progress-bar-text">
+            {/* {Math.floor(props.progress * 100)}%
+            {props.progress === 1 ? <FormattedMessage id="hub.bundle.uninstalled" /> : null} */}
+            {`${Math.floor(props.progress * 100)}%`}
+          </span>
+          {/* <p className="ComponentUninstallProcessState__in-progress-body-progress-bar-text">
```
Comment (DMinnai) [inline src/ui/component-repository/components/item/install-controls/ComponentUninstallProcessState.js]: Can we remove this ?

File: src/ui/component-repository/components/item/install-controls/ComponentUninstallProcessState.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            <span className="ComponentUninstallProcessState__in-progress-body-progress-bar-number">
+              {props.current}/{props.total}
+            </span>
+            <FormattedMessage
+              id="componentRepository.components.elementsUninstalled"
+            />
+
+          </p> */}
```
Comment (DMinnai) [inline src/ui/component-repository/components/item/install-controls/ComponentUninstallProcessState.js]: same here

File: src/ui/component-repository/components/item/install-controls/ComponentUninstallProcessState.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+      type: PropTypes.string,
+      status: PropTypes.string,
+    })),
+  }).isRequired,
+};
+
+UninstallInProgressOrFinished.defaultProps = {
+  progress: 0,
+  // current: 5,
+  // total: 6,
```
Comment (DMinnai) [inline src/ui/component-repository/components/item/install-controls/ComponentUninstallProcessState.js]: What about this props?

File: src/state/component-repository/components/actions.js [hunk 6]
```diff
@@ -399,62 +403,102 @@ export const getInstallPlan = component => dispatch => (
...
+        stepFunction: payload => stepFunction(payload.progress, payload),
       })
-      .finally(() => resolve());
-  })
-);
+        .then((response) => {
+          const {
+            payload,
+          } = response;
+          console.log('pollECRComponentUninstallStatus response', response);
```
Comment (ryanjpburgos) [inline src/state/component-repository/components/actions.js]: are the console.logs necessary?

File: src/ui/component-repository/components/list/HubBundleManagementModal.js [hunk 7]
```diff
@@ -10,17 +10,22 @@ import { getLoading } from 'state/loading/selectors';
 import { getBundleGroups, getSelectedRegistry, getSelectedBundleStatus } from 'state/component-repository/hub/selectors';
 import ComponentInstallActionsContainer from 'ui/component-repository/components/item/install-controls/ComponentInstallActionsContainer';
 import {
+  // componentUninstallOngoingProgress,
```
Comment (DMinnai) [inline src/ui/component-repository/components/list/HubBundleManagementModal.js]: What about those lines?

File: src/ui/component-repository/components/list/HubBundleManagementModal.js [hunk 8]
```diff
@@ -152,6 +158,29 @@ const HubBundleManagementModal = () => {
...
+  //     dispatch(setInstallUninstallProgress(progress));
+  //     if (newPayload && newPayload.componentId) {
+  //       dispatch(componentUninstallOngoingProgress(newPayload.componentId, newPayload));
+  //     }
+  //   };
+  //   if (component && component.code) {
+  //     dispatch(pollECRComponentCurrentUninstallJob(component.code, pollStepFunction, true));
+  //   }
+  // // eslint-disable-next-line react-hooks/exhaustive-deps
+  // }, [dispatch, component.code]);
```
Comment (DMinnai) [inline src/ui/component-repository/components/list/HubBundleManagementModal.js]: Whta about this snippet?

File: src/ui/component-repository/components/item/install-controls/ComponentUninstallManagerModal.js (hunk 1 above)
Comment (ichalagashvili) [reply src/ui/component-repository/components/item/install-controls/ComponentUninstallManagerModal.js]: yes 👍

File: src/ui/component-repository/components/item/install-controls/ComponentUninstallManagerModal.js (hunk 2 above)
Comment (ichalagashvili) [reply src/ui/component-repository/components/item/install-controls/ComponentUninstallManagerModal.js]: yes 👍

File: src/ui/component-repository/components/item/install-controls/ComponentUninstallProcessState.js (hunk 3 above)
Comment (ichalagashvili) [reply src/ui/component-repository/components/item/install-controls/ComponentUninstallProcessState.js]: yes 👍

File: src/ui/component-repository/components/item/install-controls/ComponentUninstallProcessState.js (hunk 4 above)
Comment (ichalagashvili) [reply src/ui/component-repository/components/item/install-controls/ComponentUninstallProcessState.js]: yes 👍

File: src/ui/component-repository/components/item/install-controls/ComponentUninstallProcessState.js (hunk 5 above)
Comment (ichalagashvili) [reply src/ui/component-repository/components/item/install-controls/ComponentUninstallProcessState.js]: yes 👍

File: src/ui/component-repository/components/list/HubBundleManagementModal.js (hunk 7 above)
Comment (ichalagashvili) [reply src/ui/component-repository/components/list/HubBundleManagementModal.js]: yes 👍

File: src/ui/component-repository/components/list/HubBundleManagementModal.js (hunk 8 above)
Comment (ichalagashvili) [reply src/ui/component-repository/components/list/HubBundleManagementModal.js]: yes 👍

Comment (ryanjpburgos) [inline src/ui/component-repository/components/list/HubBundleManagementModal.js]: we have a lot of commented code do we wanna keep it?

**Review by DMinnai — DISMISSED** [review]

**Review by ryanjpburgos — COMMENTED** [review]
LGTM, I just don't know if wanna keep all the console.logs

File: src/state/component-repository/components/actions.js (hunk 6 above)
Comment (ichalagashvili) [reply src/state/component-repository/components/actions.js]: I will remove those 👍

File: src/ui/component-repository/components/list/HubBundleManagementModal.js (hunk 8 above)
Comment (ichalagashvili) [reply src/ui/component-repository/components/list/HubBundleManagementModal.js]: all commented code is removed now 👍

File: src/state/component-repository/components/actions.js (hunk 6 above)
Comment (ichalagashvili) [reply src/state/component-repository/components/actions.js]: done ✅

**Review by ryanjpburgos — APPROVED** [review]

**Pull request merged**